builtin/providers/cloudstack/resource_cloudstack_vpc.go

```diff
@@ -152,6 +157,16 @@ func resourceCloudStackVPCRead(d *schema.ResourceData, meta interface{}) error {
 	setValueOrUUID(d, "project", v.Project, v.Projectid)
 	setValueOrUUID(d, "zone", v.Zonename, v.Zoneid)
 
+	// Grab the source NAT IP that CloudStack assigned.
+	p := cs.Address.NewListPublicIpAddressesParams()
+	p.SetVpcid(d.Id())
+	p.SetIssourcenat(true)
+	p.SetProjectid(v.Projectid)
```

Comment:
@hany is this completely save? The zero value of and int is `0`, so what your doing here is either setting the given project ID, or `0`. Not sure, but I guess that could mean trouble.

I guess adding a check like this would make sense:

```go
if project, ok := d.GetOk("project"); ok {
    p.SetProjectid(v.Projectid)
}
```